### PR TITLE
fix: handle redelegation ack

### PR DESCRIPTION
**Pull request by ajansari95**
## 1. Summary
Fixes # (issue)
- ensure the redelegation msgs are unique
- handle for GC redelegations in case of succesful `MsgBeginRedelegate`

## 2.Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## 4. How to test/use
- `TestRemoveDuplicates`
- `TestReceiveAckForBeginRedelegate`

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: x/interchainstaking/keeper/keeper.go [hunk 1]
```diff
@@ -523,7 +523,23 @@ type RebalanceTarget struct {
...
+		key := fmt.Sprintf("%v-%s-%s", r.Amount.String(), r.Source, r.Target)
+		if !encountered[key] {
+			encountered[key] = true
+			result = append(result, r)
+		}
+	}
+	return result
+}
+
+func DetermineAllocationsForRebalancing(currentAllocations map[string]math.Int, currentSum math.Int, targetAllocations types.ValidatorIntents, existingRedelegations []types.RedelegationRecord) RebalanceTargets {
```
Comment (aljo242) [inline x/interchainstaking/keeper/keeper.go:542]: Should this function just always return the rebalance targets with the duplicates removed? 

@joe-bowman @ajansari95

Comment (ajansari95) [reply x/interchainstaking/keeper/keeper.go:542]: yes

Comment (joe-bowman) [reply x/interchainstaking/keeper/keeper.go:542]: Yes, the logic is completely changed in v1.4, so this is just a workaround to make do until then.

**Review by joe-bowman — APPROVED** [review]
lgtm

**Review by aljo242 — APPROVED** [review]

**Pull request merged**